block_iagora.php

```diff
@@ -47,35 +47,43 @@ public function get_content() {
         }
 
         $this->content = new stdClass();
-        $this->content->footer = '';
-
         $iframeurl = isset($this->config->iframeurl) ? $this->config->iframeurl : '';
-
         if (empty($iframeurl)) {
             $this->content->text = get_string('noiframeurl', 'block_iagora');
         } else {
             $this->content->text = '<iframe src="' . $iframeurl . '" width="100%" height="400px" frameborder="0"></iframe>';
         }
-
+        $this->content->text = $this->generate_chat_content($iframeurl);
         return $this->content;
     }
-
     /**
-     * Define in which pages this block can be added.
+     * Generate the HTML content for the chat block.
      *
-     * @return array of the pages where the block can be added.
+     * @param string $iframeurl The URL to be used in the iframe.
+     * @return string The generated HTML content for the chat.
      */
-    public function applicable_formats() {
-        return ['all' => true];
+    private function generate_chat_content($iframeurl) {
+        global $PAGE, $OUTPUT;
+        // Generate a unique identifier for the chat container
+        $chat_id = uniqid('iagora_chat_');
+        // Chat button icon (using Moodle icons)
+        $chat_icon = $OUTPUT->pix_icon('t/message', get_string('openchat', 'block_iagora'));
+        $context = [
+            'chat_id' => $chat_id,
+            'chat_icon' => $chat_icon,
+            'tokenEndpointURL' => $iframeurl 
+        ];
+    
+        return $OUTPUT->render_from_template('block_iagora/chat', $context);
     }
 
     /**
-     * Allow the block to be configured.
+     * Allow the block to be configure.
      *
      * @return bool
      */
-    public function instance_allow_config() {
-        return true;
+    public function intance_allow_config() {
```

Comment:
Why do we need to change this function docstring/name/body?

Reply:
Done